include/revng/Model/Importer/Binary/BinaryImporter.h

```diff
@@ -16,9 +16,16 @@ class ObjectFile;
 } // namespace llvm
 
 struct ImporterOptions;
-llvm::Error importBinary(TupleTree<model::Binary> &Model,
-                         llvm::object::ObjectFile &BinaryHandle,
-                         const ImporterOptions &Options);
-llvm::Error importBinary(TupleTree<model::Binary> &Model,
-                         llvm::StringRef Path,
-                         const ImporterOptions &Options);
+llvm::Error
+importBinary(TupleTree<model::Binary> &Model,
+             llvm::object::ObjectFile &BinaryHandle,
+             const ImporterOptions &Options,
+             TupleTreeReference<model::BinaryIdentifier, model::Binary>
```

Comment:
Use `BinaryReference` everywhere you can.

Reply:
That wold require having a header with `using BinaryReference = TupleTreeReference<model::BinaryIdentifier, model::Binary>`. I was under the impression that `using`s in header were prohibited, did that change?

Reply:
`using`s are not prohibited in headers. `using namespace` are prohibited.